numba/np/arrayobj.py

```diff
@@ -1619,6 +1619,43 @@ def impl(*args):
     return impl
 
 
+def raise_with_shape_context(src_shapes, index_shape):
+    """Targets should implement this if they wish to specialize the error
+    handling/messages. The overload implementation takes two tuples as arguments
+    and should raise a ValueError."""
+    raise NotImplementedError
+
+
+@overload(raise_with_shape_context, target="generic")
+def ol_raise_with_shape_context_generic(src_shapes, index_shape):
+    # This overload is for a "generic" target, which makes no assumption about
+    # the NRT or string support, but does assume exceptions can be raised.
+    if ((isinstance(src_shapes, types.BaseTuple) and
+         isinstance(index_shape, types.BaseTuple))):
+        def impl(src_shapes, index_shape):
+            raise ValueError("cannot assign slice from input of different size")
+        return impl
+
+
+@overload(raise_with_shape_context, target="CPU")
+def ol_raise_with_shape_context_cpu(src_shapes, index_shape):
+    if ((isinstance(src_shapes, types.BaseTuple) and
+         isinstance(index_shape, types.BaseTuple))):
```

Comment:
```suggestion
    if ((isinstance(src_shapes, types.UniTuple) and
         isinstance(index_shape, types.UniTuple)) and
         src_shapes.dtype == index_shape.dtype and
         isinstance(src_shapes.dtype, types.Integer) ):
```
to be extra pedantic.

Reply:
Fixed in 936fa62